### PR TITLE
arvdias/use fetch func for utilization calc

### PR DESCRIPTION
**Pull request by arvdias**

### CODE REVIEW

**Comment by roobre** [comment]
@arvdias Is this related to https://github.com/newrelic/nri-kubernetes/issues/75?

File: src/metric/populate_test.go [hunk 1]
```diff
@@ -4,6 +4,8 @@ import (
 	"errors"
 	"testing"
 
+	"k8s.io/apimachinery/pkg/version"
```
Comment (invidian) [inline src/metric/populate_test.go:7]: This change seems unrelated.

File: src/metric/definition.go [hunk 2]
```diff
@@ -1006,17 +1029,21 @@ func toComplementPercentage(desiredMetric, complementMetric string) definition.F
...
 	return func(groupLabel, entityID string, groups definition.RawGroups) (definition.FetchedValue, error) {
-		dividend, err := definition.FromRaw(dividendMetric)(groupLabel, entityID, groups)
+		dividendValue, err := dividend(groupLabel, entityID, groups)
+		if err != nil {
+			return nil, err
+		}
+		divisorValue, err := divisor(groupLabel, entityID, groups)
 		if err != nil {
-			return nil, fmt.Errorf("'%s' is nil", dividendMetric)
+			return nil, err
```
Comment (invidian) [inline src/metric/definition.go:1040]: Returned errors should be wrapper to make debugging easier.

File: src/metric/definition.go [hunk 3]
```diff
@@ -1006,17 +1029,21 @@ func toComplementPercentage(desiredMetric, complementMetric string) definition.F
...
 		}
-		divisor, err := definition.FromRaw(divisorMetric)(groupLabel, entityID, groups)
+		v, err := computePercentage(dividendValue.(uint64), divisorValue.(uint64))
```
Comment (invidian) [inline src/metric/definition.go:1042]: Why is extra error checking needed here?

File: src/metric/definition.go [hunk 4]
```diff
@@ -530,9 +530,10 @@ var KSMSpecs = definition.SpecGroups{
 			{Name: "daemonsetName", ValueFunc: prometheus.FromLabelValue("kube_daemonset_created", "daemonset"), Type: sdkMetric.ATTRIBUTE},
 			{Name: "label.*", ValueFunc: prometheus.InheritAllLabelsFrom("daemonset", "kube_daemonset_labels"), Type: sdkMetric.ATTRIBUTE},
 			//computed
-			{Name: "podsMissing", ValueFunc: Subtract(
-				definition.Transform(prometheus.FromValue("kube_daemonset_status_desired_number_scheduled"), fromPrometheusNumeric),
-				definition.Transform(prometheus.FromValue("kube_daemonset_status_number_ready"), fromPrometheusNumeric)),
+			{Name: "podsMissing",
+				ValueFunc: Subtract(
```
Comment (invidian) [inline src/metric/definition.go:534]: It seems this line could be kept as it was.

File: src/metric/definition.go [hunk 5]
```diff
@@ -905,10 +906,22 @@ var KubeletSpecs = definition.SpecGroups{
...
 
 			// computed
-			{Name: "cpuCoresUtilization", ValueFunc: toUtilization("cpuUsedCores", "cpuLimitCores"), Type: sdkMetric.GAUGE},
-			{Name: "requestedCpuCoresUtilization", ValueFunc: toUtilization("cpuUsedCores", "cpuRequestedCores"), Type: sdkMetric.GAUGE},
-			{Name: "memoryUtilization", ValueFunc: toUtilization("memoryUsedBytes", "memoryLimitBytes"), Type: sdkMetric.GAUGE},
-			{Name: "requestedMemoryUtilization", ValueFunc: toUtilization("memoryUsedBytes", "memoryRequestedBytes"), Type: sdkMetric.GAUGE},
+			{
+				Name:      "cpuCoresUtilization",
+				ValueFunc: toUtilization(definition.FromRaw("cpuUsedCores"), definition.FromRaw("cpuLimitCores")),
+				Type:      sdkMetric.GAUGE},
```
Comment (invidian) [inline src/metric/definition.go:912]: To be consistent with general styling and other changes.
```suggestion
				Type:      sdkMetric.GAUGE
			},
```

**Comment by paologallinaharbur** [comment]
Closing for now because it is becoming outdated, feel free to reopen it if you want to contribute againn :)
